### PR TITLE
Feat: Cloud connections

### PR DESCRIPTION
**Pull request by tuul-wq**
- Request data from CloudStorage for previously saved connections
- Update CloudStorage when asset is connected/disconnected

= [Task](https://app.clickup.com/t/869502m30) =

### CODE REVIEW

File: app/common/utils/chains.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-export const DEFAULT_CONNECTED_CHAINS: Record<ChainId, boolean> = {
-  '0x91b171bb158e2d3848fa23a9f1c25182fb8e20313b2c1eb49219da7a70ce90c3': true, // POLKADOT
-  '0xb0a8d493285c2df73290dfb7e61f870f17b41801197a149ca93654499ea3dafe': true, // KUSAMA
-  '0x68d56f15f85d3136970ec16946040bc1752654e906147f7e43e9d539d7c3de2f': true, // DOT_ASSET_HUB
-  '0xe143f23803ac50e8f6f8e62695d1ce9e4e1d68aa36c1cd2cfd15340213f3423e': true, // WESTEND
+export const DEFAULT_CONNECTED_CHAINS: Record<ChainId, AssetId[]> = {
```
Comment (tuul-wq) [inline app/common/utils/chains.ts:1]: In case we have more than one asset to connect

File: app/models/network/network-model.test.ts [hunk 2]
```diff
@@ -26,121 +27,144 @@ const mockedChains = [
...
+      ],
+    });
 
     await allSettled(networkModel.input.networkStarted, { scope, params: 'chains_dev' });
-    expect(fakePopulateFx).toHaveBeenCalledOnce();
+    expect(fakeRequestFx).toHaveBeenCalledOnce();
     expect(scope.getState(networkModel._internal.$chains)).toEqual(mockedChainsMap);
   });
 
-  test('should add metadata subscription after createProviderFx effect', async () => {
```
Comment (tuul-wq) [inline app/models/network/network-model.test.ts:60]: Pretty useless test

File: app/models/network/network-model.test.ts [hunk 3]
```diff
@@ -26,121 +27,144 @@ const mockedChains = [
...
-    networkModel._internal.requestMetadataFx.use(() => Promise.resolve(mockMetadata));
-
     await allSettled(networkModel._internal.requestMetadataFx, { scope, params: {} as ApiPromise });
 
     expect(scope.getState(networkModel._internal.$metadata)).toEqual([mockMetadata]);
   });
 
   test('should connect to default_chains on networkStarted event', async () => {
-    const scope = fork();
+    vi.spyOn(chainsApi, 'getChainsData').mockResolvedValue(mockedChains);
```
Comment (tuul-wq) [inline app/models/network/network-model.test.ts:96]: Mock API in order to make sure effect logic works as intended

File: app/models/network/network-model.ts [hunk 4]
```diff
@@ -173,16 +182,30 @@ sample({
 });
 
 sample({
-  clock: combineEvents([populateChainsFx.doneData, getConnectedAssetsFx.doneData]),
+  clock: requestChainsFx.doneData,
```
Comment (tuul-wq) [inline app/models/network/network-model.ts:185]: Connect default chains as soon as they come, not waiting for CloudStorage

**Review by johnthecat — APPROVED** [review]

**Review by sokolova-an — APPROVED** [review]

**Pull request merged**